docs/reference/analysis/tokenizers/edgengram-tokenizer.asciidoc

```diff
@@ -209,11 +209,15 @@ The above example produces the following terms:
 ---------------------------
 
 Usually we recommend using the same `analyzer` at index time and at search
-time. In the case of the `edge_ngram` tokenizer, the advice is different.  It
+time. In the case of the `edge_ngram` tokenizer, the advice is different. It
 only makes sense to use the `edge_ngram` tokenizer at index time, to ensure
-that partial words are available for matching in the index.  At search time,
+that partial words are available for matching in the index. At search time,
 just search for the terms the user has typed in, for instance: `Quick Fo`.
 
+We also recommend using the <<analysis-truncate-tokenfilter,`truncate` token
+filter>> with a custom search analyzer to truncate tokens to the `max_gram`
+character length. Otherwise, searches for longer terms could return no results.
```

Comment:
I looked at the issue briefly yesterday and think while the truncate filter helps with avoiding empty results, it creates other problems. When doing this, we now get the same result set for each search term that exceeds the truncate length if the prefix matches. This might be unexpected for users looking for a true search-as-you-type functionality as well.

More generally speaking: if max_gram and truncate length are set to 5 and we have terms "abcdefgh" and "abcdexyz" in the index, searching for anything up to "abcde" will return both docs as expected, but e.g. searching for "abcdexy" will also return both documents, which might be a bit odd. I'm not against adding this filter here, but we should probably explain this trade-off. I'm not sure what a better solution would be tbh. other than pointing to the new "search_as_you_type" datatype which hopefully does a better job with this.

Reply:
Thanks for taking a look @cbuescher. I'll amend the docs further to better explain the tradeoffs.